tasks/install-matlab/v1/src/matlab.ts

```diff
@@ -70,18 +110,39 @@ async function resolveLatest(): Promise<string> {
     });
 }
 
-export async function setupBatch(platform: string): Promise<void> {
-    const batchInstallDir = script.defaultInstallRoot(platform, "matlab-batch");
-    const exitCode = await script.downloadAndRunScript(platform, "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v0/install.sh", batchInstallDir);
+export async function setupBatch(platform: string, architecture: string) {
+    if (architecture !== "x64") {
+        return Promise.reject(Error(`This action is not supported on ${platform} runners using the ${architecture} architecture.`));
+    }
 
-    if (exitCode !== 0) {
-        return Promise.reject(Error(`Failed to install matlab-batch. Script exited with non-zero code ${exitCode}.`));
+    const matlabBatchRootUrl: string = "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v1/";
+    let matlabBatchUrl: string;
+    let matlabBatchExt: string = "";
+    switch (platform) {
+        case "win32":
+            matlabBatchExt = ".exe";
+            matlabBatchUrl = matlabBatchRootUrl + "win64/matlab-batch.exe";
+            break;
+        case "linux":
+            matlabBatchUrl = matlabBatchRootUrl + "glnxa64/matlab-batch";
+            break;
+        case "darwin":
+            matlabBatchUrl = matlabBatchRootUrl + "maci64/matlab-batch";
+            break;
+        default:
+            return Promise.reject(Error(`This action is not supported on ${platform} runners.`));
     }
 
+    const tempPath = await downloadToolIfNecessary(matlabBatchUrl, `matlab-batch${matlabBatchExt}`);

```

Comment:
This was a workaround I was doing because Azure fails the download if the file already exists. I wasn't super worried because it's downloading to runner temp, but I'm not sure how long that persists.

I'll change it to remove the file if it exists then download.

Reply:
Yeah it will. Same with the download for MATLAB. That's why I think we're not "all the way there" on self-hosted workflows. We haven't thought them all the way through or bashed them enough.

Reply:
Oh that's right, this is in `InstallMATLAB` and not `RunMATLABCommand`. Cool.

Maybe all places `downloadToolIfNecessary` is used should be changed to `downloadAndOverwriteTool` or just `downloadTool` (and the overwrite is implied). I think the risk of the "if necessary" is that a corrupt or partially downloaded version of the tool is stuck on disk and we keep trying to use it instead of getting a new one.

Reply:
updated